### DISCUSSION TITLE
Is it Possible two maintain two directories(which i have to publish in Git Hub Pages) in single repo

### DISCUSSION THREAD

**Discussion by manoj0222** [community/community#123375] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello everyone,

My project folders are located in a repository that I update on a daily basis. Let's see, I have three distinct project folders in this one repository, and I want to use git hub pages to publish these three projects. Is it feasible? If so, how is it achievable?   
                                                                   
Additionally, suppose I had a repository with two directories: one for frontend (React.js) technology and the other for backend (Node.js). How should i publish these?

**Comment by CiceroLino** [comment]
Maybe with github actions (I never tried), but you can merge frontend with backend if you dont know. It's called monolith project.

But I dont know how works your project so, the best you can do now it's make repository separetely

**Comment by SYNAX07** [comment]
To publish multiple projects using GitHub Pages from a single repository, you have a couple of options depending on the structure of your repository.

### Publishing Three Project Folders

If you have three distinct project folders within one repository and you want to publish each project separately using GitHub Pages, you can follow these steps:

1. **Branches for Each Project**:
   - Create a new branch in your repository for each project. For example, `project1`, `project2`, and `project3`.
   - Move the content of each project into its respective branch. Each project should be in the root directory of its branch.

2. **Configure GitHub Pages**:
   - Go to your repository on GitHub.
   - Navigate to the "Settings" tab.
   - Scroll down to the "GitHub Pages" section.
   - In the "Source" dropdown, select the branch corresponding to the project you want to publish (e.g., `project1` branch for Project 1).
   - Choose the root folder of the project (e.g., `/`).

3. **Publishing**:
   - GitHub Pages will now publish the selected branch and directory. Each project will have its own GitHub Pages URL based on the selected branch.

### Publishing Frontend and Backend Separately

For a repository with frontend (React.js) and backend (Node.js) directories:

1. **Separate Repositories**:
   - Consider splitting your frontend and backend into separate repositories if they represent distinct projects.

2. **Single Repository** (Subdirectories Approach):
   - If you prefer to keep them in one repository for convenience:
     - Ensure your repository has a root directory for each (`frontend/` and `backend/`).
     - Configure GitHub Pages to publish only the frontend or backend directory, depending on what you want to showcase.

3. **Steps**:
   - Create appropriate branches for frontend and backend (e.g., `frontend` and `backend`).
   - Set up GitHub Pages for each branch, pointing to the respective directory (`/frontend` or `/backend`).
   - Use separate URLs to access frontend and backend.

### Example Setup

Suppose you have a repository structured like this:

```
- ProjectRepository
  - project1
  - project2
  - project3
  - frontend
  - backend
```

For publishing:
- Create branches like `project1`, `project2`, `project3`, `frontend`, and `backend`.
- Configure GitHub Pages to publish each branch from its respective directory.

Remember, GitHub Pages can publish static content. If your backend is serving dynamic content through Node.js, consider other hosting solutions (like Heroku) for backend deployment while using GitHub Pages for frontend if needed.

**Reply by manoj0222** [reply]
<img width="519" alt="GitHub" src="https://github.com/community/community/assets/100223762/d36ec7cb-bcfe-4de3-92e2-197fdcafbf33">




I understood the maintenance part of branch and publishing part.

Can you share the folder structure for inside project1, project2.Also i had attached visual represent diagram for more easier, if would you like means you can modify the same or just share an example.

example URL:[https://app.eraser.io/workspace/b96TrzSa8WI4H6jBFjqc?origin=share].
